core/src/replay_stage.rs

```diff
@@ -450,6 +457,44 @@ impl ReplayStage {
         )
     }
 
+    fn verify_vote_and_reset_bank(
+        unverified_blocks: &RwLock<UnverifiedBlocks>,
+        vote_bank: &Option<Arc<Bank>>,
+        reset_bank: &Option<Arc<Bank>>,
+    ) -> bool {
+        if let Some(vote_bank) = vote_bank {
+            assert_eq!(vote_bank.slot(), reset_bank.as_ref().unwrap().slot());
+            // Verify `vote_bank`'s ancestors and `vote_bank`
+            if let Some(res) = unverified_blocks
+                .read()
+                .unwrap()
+                .is_verified(vote_bank.slot())
+            {
+                res
+            } else {
+                // TODO: Wait on channel for result of verification of
```

Comment:
here and in similar case below, replay is waiting on the results of the verification, we can either poll or use a channel for notification